board/drivers/uart.h

```diff
@@ -14,7 +14,7 @@ typedef struct uart_ring {
 
 void uart_init(USART_TypeDef *u, int baud);
 
-int getc(uart_ring *q, char *elem);
+bool getc(uart_ring *q, char *elem);
 int putc(uart_ring *q, char elem);
```

Comment:
bool?

Reply:
seems like it returns 1 when it succeeds and zero when it does not.
```
int getc(uart_ring *q, char *elem) {
  int ret = 0;

  enter_critical_section();
  if (q->w_ptr_rx != q->r_ptr_rx) {
    if (elem != NULL) *elem = q->elems_rx[q->r_ptr_rx];
    q->r_ptr_rx = (q->r_ptr_rx + 1) % FIFO_SIZE;
    ret = 1;
  }
  exit_critical_section();

  return ret;
}

```